aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -587,6 +632,75 @@ public AWSMobileClient getEscapeHatch() {
         return awsMobileClient;
     }
 
+    private void signInWithWebUIHelper(
+        @Nullable AuthProvider authProvider,
+        @NonNull Activity callingActivity,
+        @NonNull AuthWebUISignInOptions options,
+        @NonNull Consumer<AuthSignInResult> onSuccess,
+        @NonNull Consumer<AuthException> onException
+    ) {
+        HostedUIOptions.Builder optionsBuilder = HostedUIOptions.builder();
+
+        if (options != null) {
+            if (options.getScopes() != null) {
+                optionsBuilder.scopes(options.getScopes().toArray(new String[options.getScopes().size()]));
+            }
+
+            if (!options.getSignInQueryParameters().isEmpty()) {
+                optionsBuilder.signInQueryParameters(options.getSignInQueryParameters());
+            }
+
+            if (!options.getSignOutQueryParameters().isEmpty()) {
+                optionsBuilder.signOutQueryParameters(options.getSignOutQueryParameters());
+            }
+
+            if (!options.getTokenQueryParameters().isEmpty()) {
+                optionsBuilder.tokenQueryParameters(options.getTokenQueryParameters());
+            }
+
+            if (options instanceof AWSCognitoAuthWebUISignInOptions) {
+                AWSCognitoAuthWebUISignInOptions cognitoOptions = (AWSCognitoAuthWebUISignInOptions) options;
+                optionsBuilder.idpIdentifier(cognitoOptions.getIdpIdentifier())
```

Comment:
would it make sense to have `HostedUIOptions#fromAuthWebUISignInOptions()` method to copy over values?

Reply:
HostedUIOptions is a class in AWSMobileClient so I can't modify it and this is the only place this functionality is needed so I don't think it's an instance where we should separate it out into a helper method either.